### ISSUE TITLE
Automerge may attach deleted mark to new inserted text

### ISSUE THREAD

**Issue by zxch3n** [automerge/automerge#604]
Steps to reproduce:

```rust
#[test]
fn mark() {
    let mut doc = Automerge::new();
    let mut tx = doc.transaction();
    let text_id = tx.put_object(&ROOT, "text", ObjType::Text).unwrap();
    tx.splice_text(&text_id, 0, 0, "hello world").unwrap();
    let mark = Mark::new("bold".to_string(), true, 2, 8);
    tx.mark(&text_id, mark, ExpandMark::After).unwrap();
    let mark = Mark::new("link".to_string(), true, 3, 6);
    tx.mark(&text_id, mark, ExpandMark::None).unwrap();

    tx.splice_text(&text_id, 1, 10, "").unwrap(); // 'h'
    dbg!(tx.text(&text_id).unwrap(), tx.marks(&text_id).unwrap());
    tx.splice_text(&text_id, 0, 0, "a").unwrap(); // 'ah'
    dbg!(tx.text(&text_id).unwrap(), tx.marks(&text_id).unwrap());
    tx.splice_text(&text_id, 2, 0, "a").unwrap(); // 'ah<bold>a</bold>'
    dbg!(tx.text(&text_id).unwrap(), tx.marks(&text_id).unwrap());
}
```

Before the last insertion, the text is `ah` with no mark. So the expected marks after the last insertion should be empty as well. But actually the inserted 'a' has a bold mark.

It's caused by the Peritext algorithm. I describe the suggested fix here https://github.com/inkandswitch/peritext/issues/32

**Comment by orionz** [comment]
Found a good fix for this.  Basically when scanning tombstones ignore all mark begin/end pairs or marks between begin/end pairs

**Comment by alexjg** [comment]
Fixed by #629
